tools/slang-capability-generator/capability-generator-main.cpp

```diff
@@ -64,8 +87,11 @@ struct CapabilityDef : public RefObject
     List<List<CapabilityDef*>> canonicalRepresentation;
     SerializedArrayView serializedCanonicalRepresentation;
     SourceLoc sourceLoc;
+    HashSet<const CapabilityDef*> abstractAtomsPresent;
```

Comment:
What is the purpose of this member, can you put a comment here?

Reply:
The naming here is still bit confusing, since `glsl` is not an abstract atom, but a concrete atom that inherits from an abstract atom. The set here probably refers to things like `glsl` and not `target`, right?

Reply:
> The set here probably refers to things like glsl and not target, right?
yes.

I don't really know a good word to simplify "list of 'stage' and 'target' abstract typed atoms"